### PR TITLE
Enhanced weather support (including overlay)

### PR DESCRIPTION
**Pull request by CalamityJames**
**Obsoletes #46.**

## What it does differently
- Uses multiple sprite sheets instead of using ImageMagick to generate static icons (yes, it **was** a terrible idea).
- Provides a weather overlay using `s2geometry.js` to convert the S2 cells from your weather table into polygons
- When `zoom > 13`, the weather cell icon moves to the header
- You can toggle the weather overlay in the "Locations & Settings" section of the sidebar

## New config options
```
$noWeatherOverlay = false;                                          // true/false
$enableWeatherOverlay = 'false';                                    // true/false
```
This controls the appearance of the "Weather Conditions" toggle switch within "Location & Search Settings" (todo: move these into a new tab of the sidebar)

## Spritesheet info
No need for ImageMagick any more, we now use multiple sprite sheets (one for each weather condition). These are located in `static/sprites-safe/`

### Non-safe sprites
If you use non-safe sprites, please DM me for more info on how to use these.

## RocketMap
I have not yet tested this fork as I use Monocle-Alternative however after looking at the code, I believe I have added support for their PR 2420. I will need users to test and confirm this though.

## Update drill
Two different options:
### If coming from #46
Sorry, it's all changed, time to revert!
- Revert to `develop`
- Pull this PR
- `npm install`
- `npm run build`
### If coming from `develop`
- Pull this PR
- `npm install`
- `npm run build`

## nginx users
You'll need an extra rewrite in your config file:
```
    rewrite ^/weather_data$ /weather_data.php?$1 last;
```

## Known issues
Spawns will not update if weather changes

## Suggestions
As always, I'm open to suggestions on how to make this more efficient, so if you have any, please let me know!

### CODE REVIEW

**Comment by Falaris01** [comment]
Totally loving it, as do my users!

Only issue some are having on small phone screens is that the weather icon gets moved from the top bar to the top right corner of the map and is blocking the X of the gymsidebar, also the stats toggle disappears. Happens when the Title is too long, e. g. a short Map Name, Discord link, Paypal link is already too much.

Maybe moving it under the fullscreen toggle would be a nice workaround?

**Comment by CalamityJames** [comment]
@Falaris01 I pushed a commit to fix the bug you reported this morning :) however I agree it should probably go somewhere else - I will do some testing and see where works

**Comment by Falaris01** [comment]
I see now that I was a commit behind, sorry about that!

File: weather_data.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
+<?php
+include('config/config.php');
+global $map, $fork;
+header('Content-Type: application/json');
+// init map
+if (strtolower($map) == "monocle") {
```
Comment (hammydown4325) [inline weather_data.php]: Only thing i would do is convert == to ===

File: static/js/map.js [hunk 2]
```diff
@@ -90,14 +92,20 @@ var onlyTriggerGyms
...
 
 
 var genderType = ['♂', '♀', '⚲']
 var unownForm = ['unset', 'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z', '!', '?']
 var cpMultiplier = [0.094, 0.16639787, 0.21573247, 0.25572005, 0.29024988, 0.3210876, 0.34921268, 0.37523559, 0.39956728, 0.42250001, 0.44310755, 0.46279839, 0.48168495, 0.49985844, 0.51739395, 0.53435433, 0.55079269, 0.56675452, 0.58227891, 0.59740001, 0.61215729, 0.62656713, 0.64065295, 0.65443563, 0.667934, 0.68116492, 0.69414365, 0.70688421, 0.71939909, 0.7317, 0.73776948, 0.74378943, 0.74976104, 0.75568551, 0.76156384, 0.76739717, 0.7731865, 0.77893275, 0.7846369, 0.79030001]
 
+var weatherArray = []
+var weatherPolys = []
+var weatherMarkers = []
+var weatherColors = ['grey', '#fdfd96', 'darkblue', 'grey', 'darkgrey', 'purple', 'white', 'black']
```
Comment (hammydown4325) [inline static/js/map.js]: Any reason we couldn't move this to the config for people to configure colors that they want? I've had the request from a couple different mappers just seems like a convenience.

**Review by hammydown4325 — CHANGES_REQUESTED** [review]
Just some small little quality of life improvements.

**Comment by CalamityJames** [comment]
@iroken222 Good suggestions, thanks. Implemented :)

File: lib/RocketMap_Sloppy.php [hunk 3]
```diff
@@ -4,5 +4,67 @@
...
+        }
+        if (!empty($minLevel) && !is_nan((float)$minLevel) && $minLevel != 0) {
+            if (empty($exMinIv)) {
+                $conds[] = 'cp_multiplier >= ' . $this->cp_multiplier[$minLevel];
+            } else {
+                $conds[] = '(cp_multiplier >= ' . $this->cp_multiplier[$minLevel] . ' OR pokemon_id IN(' . $exMinIv . ') )';
+            }
+        }
+        return $this->query_active($select, $conds, $params);
+    }
```
Comment (Glennmen) [inline lib/RocketMap_Sloppy.php:119]: I think you forgot `get_active_by_id()` ;)

Comment (CalamityJames) [reply lib/RocketMap_Sloppy.php:119]: Good spot, fixed :)

**Pull request merged**